src/query.rs

```diff
@@ -214,3 +230,50 @@ pub fn q_once<'sqlite, 'schema, 'query, T>
 
     run_algebrized_query(sqlite, algebrized)
 }
+
+pub fn q_explain<'sqlite, 'schema, 'query, T>
+(sqlite: &'sqlite rusqlite::Connection,
+ schema: &'schema Schema,
+ query: &'query str,
+ inputs: T) -> Result<QueryExplanation>
+        where T: Into<Option<QueryInputs>>
+{
+    let parsed = parse_find_string(query)?;
+    let algebrized = algebrize_with_inputs(schema, parsed, 0, inputs.into().unwrap_or(QueryInputs::default()))?;
+
+    let unbound = algebrized.unbound_variables();
+    if !unbound.is_empty() {
+        bail!(ErrorKind::UnboundVariables(unbound.into_iter().map(|v| v.to_string()).collect()));
+    }
+
+    let select = query_to_select(algebrized)?;
+    let query = select.query.to_sql_query()?;
+
+    let plan_sql = format!("EXPLAIN QUERY PLAN {}", query.sql);
+
+    let mut statement = sqlite.prepare(plan_sql.as_str())?;
+
+    let from_row = |row: &rusqlite::Row| {
+        QueryPlanStep {
+            select_id: row.get(0),
+            order: row.get(1),
+            from: row.get(2),
+            detail: row.get(3)
+        }
+    };
+
+    let mut rows = if query.args.is_empty() {
+        statement.query_map(&[], from_row)?
+    } else {
+        let refs: Vec<(&str, &ToSql)> =
+            query.args.iter()
+                .map(|&(ref k, ref v)| (k.as_str(), v.as_ref() as &ToSql))
+                .collect();
+        statement.query_map_named(refs.as_slice(), from_row)?
+    };
+    let mut steps = Vec::new();
+    while let Some(step) = rows.next() {
```

Comment:
One of the reasons for using `query_map` is that it returns a `MappedRows`, which implements `Iterator`. Rather than this manual collection, you can just do `let steps: Vec<QueryPlanStep> = rows.collect();`.

Reply:
Each item is a result (note the `?` on line 276), which is why I did it this way. I'll see if I can avoid the explicit iteration without making it too complex though.